packages/app/src/source.ts

```diff
@@ -82,17 +100,19 @@ export function useSources() {
         return getLocalSources();
       }
 
-      return hdxServer('sources').json<TSource[]>();
+      const rawSources = await hdxServer('sources').json<TSourceUnion[]>();
+      return rawSources.map(addDefaultsToSource);
     },
   });
 }
 
 export function useSource({ id }: { id?: string | null }) {
   return useQuery({
     queryKey: ['sources'],
-    queryFn: async () => {
+    queryFn: async (): Promise<TSource[]> => {
```

Comment:
nit: TS should infer the return type here

Reply:
Thanks, removed